Source/Definition/Method/TyphoonMethod.m

```diff
@@ -90,9 +90,10 @@ - (NSUInteger)hash
 - (NSUInteger)calculateHash
 {
     NSUInteger hash = (NSUInteger) sel_getName(_selector);
+    NSUInteger magicNumber = 5;
 
     for (id <TyphoonParameterInjection> parameter in _injectedParameters) {
-        hash = (NSUInteger) ((5 << hash) - hash + [[parameter description] hash]);
+        hash = (NSUInteger) ((magicNumber << hash) - hash + [[parameter description] hash]);
```

Comment:
In this particular case that's well-known (at least in Java world) hash algorithm which uses 31 multiplier. If you need more details about 31, see this answer http://stackoverflow.com/a/299748

Here is optimized implementation using bit shifting, since left shifting 5 bytes means multiplying by 32
```
hash = a
hash = hash * 31 + b
hash = hash * 31 + c
```
equals
```
hash = a
hash = 5 << hash - hash + b
hash = 5 << hash - hash + c
```